hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodieClientOnCopyOnWriteStorage.java

```diff
@@ -216,6 +222,71 @@ public void testDeduplicationOnUpsert() throws Exception {
     testDeduplication(SparkRDDWriteClient::upsert);
   }
 
+  @Test
+  public void testMultiWriter() throws Exception {
```

Comment:
lets add a new test around the multi writing please. it should ideally be agnostic of COW/MOR. and we can just test MOR.